### ISSUE TITLE
Issues with Submitting Payment & Validation in Prod

### ISSUE THREAD

**Issue by johngruen** [mozilla/fxa#3106]
## TL;DR

* We return user hostile errors if stripe tells us we provide a bad CVC or Expiration Date.
* We let users pay successfully with a bad name
* We let users pay with a bad Zip.
* In none of these cases was i able to produce the state mismatch described in #3061. While these may be related, this is a separate set of issues.

## Body

There seem to be a lot of cases where this ugly/unhelpful error modal is showing up on the Payments page

<img width="708" alt="225-d8736dcac61ab60c9bd4ad45ef9ea6c9_subscription_failed" src="https://user-images.githubusercontent.com/3323249/67394483-8c5f2c80-f5a4-11e9-80ac-d3fb839aea02.png">

This error message, btw, comes from the fxa-auth-server. Here's a [link](https://github.com/mozilla/fxa/blob/1c74302f45120cb8720aa26c2be4c53ae2d3e84d/packages/fxa-auth-server/lib/error.js).

## Overview of what errors look like from Stripe

Starting with what Stripe is showing b/c it's pretty similar on all the error cases, and maybe

So It looks like when a user submits a subscription on FxA does a post request at `v1/tokens`.

The response body from Stripe has a lot of stuff in it, but importantly it includes the following lines.

```
"address_zip_check": "unchecked",
"cvc_check": "unchecked"
```

So at the time the token is created, nothing is verified on the stripe end apparently. Not sure that's important, but it's worth noting.

The next thing that three **Identical** POSTS are made `v1/customers`. You can see this pattern in the following image

![image](https://user-images.githubusercontent.com/3323249/67397126-0691b000-f5a9-11e9-88c3-76be9a500ec7.png)

Stripe returns the same response to all three (some of these responses are documented below)

### Test 1: Incorrect CVC

```
{
  "error": {
    "code": "incorrect_cvc",
    "doc_url": "https://stripe.com/docs/error-codes/incorrect-cvc",
    "message": "Your card's security code is incorrect.",
    "param": "cvc",
    "type": "card_error"
  }
}
```

#### Proposed Fix

Correctly transact Stripe's error codes back to the FxA client and display them on the appropriate fields where possible. 

### Test 2: Incorrect Date

Note, when i test an invalid date, I was a slightly different error response that's more generic. Per above note on the Stripe token API, there does not seem to be a unique date check field

```
{
  "error": {
    "code": "card_declined",
    "decline_code": "do_not_honor",
    "doc_url": "https://stripe.com/docs/error-codes/card-declined",
    "message": "Your card was declined.",
    "param": "",
    "type": "card_error"
  }
}
```
#### Proposed Fix

So, let's start with this... I hate the modal for server errors and I would rather kill it. It's a pattern that blocks interaction and you lose the message when you have to make changes to the form. I'd rather errors in Sub Plat be displayed in the form similar to other server-driven errors in FxA.

<img width="614" alt="Screen Shot 2019-10-23 at 3 13 01 PM" src="https://user-images.githubusercontent.com/3323249/67400360-d4cf1800-f5ad-11e9-8671-bd9fada507e6.png">

For now, we can just use a less hostile string in the modal.
"Your transaction could not be processed. Please verify your credit card information and try again."

## Test 3: Incorrect Name

If the user enters a name incorrectly, they can still subscribe...For example, I entered `Brosef Jenkins` as my name and was able to subscribe. looking at Stripe's `v1/customers` API response, I do not see any sort of `name_verified` property happening in Stripe so there's no returned warning to us.

### Proposed fix

Not sure, and it turns out it might be a [wontfix](https://stackoverflow.com/questions/41322875/stripe-card-holder-name-verification).

### Test 4: Incorrect ZIP

Since the ZIP code is just an input and not a Stripe iframe, i tested separately. Results here are confusing...

So i submitted a valid CC with an invalid ZIP and was able to subscribe without issue...in the `v1/customers` post response i see the following

```
"address_zip": "22222",
"address_zip_check": "fail",
...
"cvc_check": "pass"
```

I wonder if the valid CVC overrules the invalid ZIP code here. 

### Proposed Fix
I wonder. @bbangert if we could configure stripe to fail on Zip failure?

**Comment by johngruen** [comment]
@stlhood, I've gone ahead and updated our stripe settings for ZIP verification.

**Comment by johngruen** [comment]
Note...I've added a rule to check zips....

![image](https://user-images.githubusercontent.com/3323249/67405294-da7c2c00-f5b4-11e9-8c1a-13d84627d905.png)

Weirdly, the error response from Stripe is now this:
```
{
  "error": {
    "code": "incorrect_cvc",
    "doc_url": "https://stripe.com/docs/error-codes/incorrect-cvc",
    "message": "Your card's security code is incorrect.",
    "param": "cvc",
    "type": "card_error"
  }
}
```

**Comment by clouserw** [comment]
I tried to find some history on these error messages since I know we wanted to collapse some error messages to avoid some fraud cases.  @meandavejustice implemented errors in #1589 with https://docs.google.com/spreadsheets/d/1Kl2v8hJTntQvtGUTJcvWpSZQLIVvoYgnRboT2Lziu_w/edit?disco=AAAADAewemI&usp_dm=false&ts=5cdc2bc0#gid=0 as guidance.  @bsurd notes [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1590106#c2) that this same error happens for:

>     incorrect cvc
>     processing error
>     stolen card
>     lost card
>     insufficient funds
>     card declined

@marty331 wontfix'd https://github.com/mozilla/subhub/issues/98 saying we would talk about details.  There is a reference there to https://github.com/mozilla/subhub/issues/148 which is still open but should probably be prioritized now.

So, in short, this feels like a bug in FxA (maybe we're falling through an error case because we don't handle those 402s?), but we maybe can't finish it until that subhub 148 issue is fixed.

**Comment by LZoog** [comment]
FWIW, the error messaging also relates to #2812.

**Comment by johngruen** [comment]
Verified that the CVC error is fixed...
Expiry and Zip Code errors show the "back end service" message

**Comment by johngruen** [comment]
Closing in favor of the two issues above
